### PR TITLE
Create plugin that joins strings #726

### PR DESCRIPTION
**Pull request by yk-jp**
Create plugin that joins strings #726

added plugin code and test code 

----

**Please** following this checklist to help us incorporate your contribution quickly and easily:

 - [x] Give the same name to the pull request as the name of ticket 
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Test your code throughly before PR.
 
If your pull request is about ~20 lines of code add license information in your code in form of comment. i.e. # License: MIT, Author:  Your name.

To make clear that you license your contribution under the MIT you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [MIT license](https://opensource.org/licenses/MIT)

PRs without ticked MIT license will not be accepted - Sorry.

### CODE REVIEW

File: tracardi/process_engine/action/v1/strings/string_join/model/configuration.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+class Configuration(PluginConfig):
+    string: str
+    delimiter: str
+
+    @validator('string')
+    def string_must_not_be_empty(cls, value):
+        if len(value.strip()) <= 0:
+            raise ValueError('String must not be an empty')
+        return value
```
Comment (atompie) [inline tracardi/process_engine/action/v1/strings/string_join/model/configuration.py:13]: Delimiter should not also be empty, and should be single character.

File: tracardi/process_engine/action/v1/strings/string_join/plugin.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+class JoinAction(ActionRunner):
+    config: Configuration
+
+    async def set_up(self, init):
+        self.config = validate(init)
+
+    async def run(self, payload: dict, in_edge=None) -> Result:
+        dot = self._get_dot_accessor(payload)
+        string = dot[self.config.string]
+        result = self.config.delimiter.join(string)
```
Comment (atompie) [inline tracardi/process_engine/action/v1/strings/string_join/plugin.py:21]: Please check if the string is an array. If it is not raise ValueError.  And I would name id data_to_join or something like this to be more clear.

File: tracardi/process_engine/action/v1/strings/string_join/plugin.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        start=False,
+        spec=Spec(
+            module='tracardi.process_engine.action.v1.strings.string_join.plugin',
+            className='JoinAction',
+            inputs=["payload"],
+            outputs=["payload"],
+            version='0.7.2',
+            license="MIT",
+            author="Yusuke Kohatsu",
+            manual="string_join_action",
```
Comment (atompie) [inline tracardi/process_engine/action/v1/strings/string_join/plugin.py:36]: You have pointed to the documentation file and this is great. If you could place the string_join_action.md in the tracardi-api project in the folder: docs/flow/actions and fill it with the simple documentation that would be great.

**Review by atompie — CHANGES_REQUESTED** [review]
The plugin looks very nice. I see you got it all. I have some minor change requests. Please correct it and I will merge it.

File: tracardi/process_engine/action/v1/strings/string_join/plugin.py (hunk 3 above)
Comment (yk-jp) [reply tracardi/process_engine/action/v1/strings/string_join/plugin.py:40]: Will do it. Let me have time to work on please.

File: tracardi/process_engine/action/v1/strings/string_join/plugin.py (hunk 2 above)
Comment (yk-jp) [reply tracardi/process_engine/action/v1/strings/string_join/plugin.py]: It's added at c1d01e9

File: tracardi/process_engine/action/v1/strings/string_join/model/configuration.py (hunk 1 above)
Comment (atompie) [reply tracardi/process_engine/action/v1/strings/string_join/model/configuration.py:19]: No lets make it that space is also delimiter.

File: tracardi/process_engine/action/v1/strings/string_join/plugin.py (hunk 2 above)
Comment (atompie) [reply tracardi/process_engine/action/v1/strings/string_join/plugin.py]: great

File: tracardi/process_engine/action/v1/strings/string_join/plugin.py (hunk 3 above)
Comment (atompie) [reply tracardi/process_engine/action/v1/strings/string_join/plugin.py:40]: Super thanks. I will make it a separate issue. So I can merge this one.

**Review by atompie — APPROVED** [review]

**Comment by atompie** [comment]
@yk-jp Merged and open a separate documentation issue. https://github.com/Tracardi/tracardi/issues/741. Please comment on it so I can assign it to you.

**Pull request merged**